doc/contrib/api/review.md

```diff
@@ -0,0 +1,15 @@
+On potential changes (including extensions) of the API/ABI please check following list:
+
+- [ ] [use case](/doc/usecases) for API exists
+- [ ] API design [review](reviews/) template is created by author of API change
```

Comment:
What exactly does this mean? I guess the author has to copy some template files, but do have to do anything else? Should they do an initial review themselves? Should they just use the review template as guidelines when designing the API?

Since this is a checklist the items must be very clear IMHO. If something is unclear the checklist is not helpful.

Reply:
I agree, very good points. @lawli3t can you please take over the PR and extend on this (how you did the reviews etc.).